File: versions/src/main/java/wtf/zani/spice/platform/impl/forge/asm/TransformerPlugin.java
package wtf.zani.spice.platform.impl.forge.asm;
//#if FORGE

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;

import java.util.Map;

public class TransformerPlugin implements IFMLLoadingPlugin {
    @Override
    public String[] getASMTransformerClass() {
        return new String[0];
    }

    @Override
    public String getModContainerClass() {
        return null;
    }

    @Override
    public String getSetupClass() {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> map) {

    }

    @Override
    public String getAccessTransformerClass() {
        return null;
    }
}
//#endif

File: versions/src/main/kotlin/org/polyfrost/spice/platform/impl/forge/asm/TransformerPlugin.kt
package org.polyfrost.spice.platform.impl.forge.asm
//#if FORGE

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin

@IFMLLoadingPlugin.Name("Spice")
@IFMLLoadingPlugin.MCVersion("1.8.9")
class TransformerPlugin : IFMLLoadingPlugin {
    override fun getASMTransformerClass(): Array<String> {
        return arrayOf(ClassTransformer::class.java.getName())
    }

    override fun getModContainerClass(): String? {
        return null
    }

    override fun getSetupClass(): String? {
        return null
    }

    override fun injectData(map: Map<String?, Any?>) {}
    override fun getAccessTransformerClass(): String? {
        return null
    }
}

//#endif
